lib/errorUtils.js

```diff
@@ -0,0 +1,63 @@
+'use strict'
+
+const Assert = require('assert')
+
+const types = {
+  system: [
+
+      // JavaScript
+
+    EvalError,
+    RangeError,
+    ReferenceError,
+    SyntaxError,
+    TypeError,
+    URIError,
+
+      // Node
+
+    Assert.AssertionError
+  ]
+}
+
+function reflect (err, types) {
+  if (match(err, types)) {
+    toss(err)
+  }
+}
+
+function toss (err, rethrow) {
+  if (rethrow || canHandlePromiseRejection()) {
+    throw err
+  } else {
+    // fallback cause DEP0018
+    process.exit(1)
```

Comment:
I’m not convinced of exit(1). I would rather rethrow every time, and let the user use make-promises-safe or a custom handler.

Reply:
done